src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java

```diff
@@ -11,168 +10,87 @@
-        }
-        if (!properties.containsKey(SapsPropertiesConstants.FOGBOW_KEYSTONEV3_SWIFT_URL)) {
-            LOGGER.error("Required property " + SapsPropertiesConstants.FOGBOW_KEYSTONEV3_SWIFT_URL + " was not set");
-            return false;
-        }
-
-        LOGGER.debug("All properties are set");
-        return true;
+    private PermanentStorage createStorageInstance(String type) throws PermanentStorageException {
+        String lType = type.toLowerCase();
+        if (lType.equals("swift"))
+            return new SwiftPermanentStorage(properties);
+        if (lType.equals("nfs"))
+            return new NfsPermanentStorage();
+
+        throw new PermanentStorageException("Failed to recognize type of permanent storage");
     }
 
-    /**
-     * This function checks if properties for debug mode have been set.
-     *
-     * @param properties saps properties to be check
-     * @return boolean representation, true (case all properties been set) or false
-     * (otherwise)
-     */
-    private boolean checkPropertiesDebugMode(Properties properties) {
-        if (!properties.containsKey(SapsPropertiesConstants.SWIFT_FOLDER_PREFIX_DEBUG_FAILED_TASKS)) {
-            LOGGER.error("Required property " + SapsPropertiesConstants.SWIFT_FOLDER_PREFIX_DEBUG_FAILED_TASKS
-                    + " was not set (it's necessary when debug mode)");
-            return false;
-        }
-
-        LOGGER.debug("All properties for debug mode are set");
-        return true;
+    private boolean checkProperties(Properties properties) {
+        String[] propertiesSet = {
+                SapsPropertiesConstants.IMAGE_DATASTORE_IP,
+                SapsPropertiesConstants.IMAGE_DATASTORE_PORT,
+                SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_GARBAGE_COLLECTOR,
+                SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_ARCHIVER,
+                SapsPropertiesConstants.SAPS_EXPORT_PATH,
+                SapsPropertiesConstants.SAPS_PERMANENT_STORAGE_TYPE
+        };
+
+        return SapsPropertiesUtil.checkProperties(properties, propertiesSet);
     }
 
-    /**
-     * Start Archiver component
-     *
-     * @throws Exception
-     */
-    public void start() throws Exception {
+    public void start() throws ArchiverException {
         cleanUnfinishedArchivedData();
 
-        sapsExecutor.scheduleWithFixedDelay(new Runnable() {
-                                                @Override
-                                                public void run() {
-                                                    garbageCollector();
-                                                }
-                                            }, 0, Long.valueOf(properties.getProperty(SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_GARBAGE_COLLECTOR)),
+        sapsExecutor.scheduleWithFixedDelay(() -> garbageCollector(), 0, Long.valueOf(properties.getProperty(SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_GARBAGE_COLLECTOR)),
```

Comment:
As I told @wesleymonte. avoid using the config in other places than the constructor

Reply:
Done.